site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -103,33 +104,33 @@ public function uploadGradeable(): JsonResponse {
                     $values['external_repo'] = $_POST['vcs']['vcs_path'];
                 }
                 if (isset($_POST['vcs']['vcs_subdirectory'])) {
-                    $values['using_subdirectory'] = 'true';
+                    $values['using_subdirectory'] = true;
                     $values['vcs_subdirectory'] = $_POST['vcs']['vcs_subdirectory'];
                 }
-                $values['vcs'] = 'true';
+                $values['vcs'] = true;
                 $values['vcs_radio_buttons'] = $_POST['vcs']['repository_type'];
                 $values['vcs_path'] = $_POST['vcs']['vcs_path'];
             }
-            $values['bulk_upload'] = (bool) $_POST['bulk_upload'] ? 'true' : 'false';
+            $values['bulk_upload'] = $_POST['bulk_upload'] ?? false;
```

Comment:
Does this need to be cast, or are you relying on conversions taking place later on?  The same applies elsewhere as well.

```suggestion
            $values['bulk_upload'] = (bool) ($_POST['bulk_upload'] ?? false);
```

Reply:
I guess it's preference, I don't really see a need to cast it here, but if you think it would be better, then we can. 
I was under the assumption that the conversion could be left to the function later, as the variables are expected to be actual booleans, but if someone uses boolean strings, it would accept them also.